npm/design-system/src/library/hooks/useFuzzySearch.ts

```diff
@@ -0,0 +1,74 @@
+import Fuse from 'fuse.js'
+import { useEffect, useLayoutEffect, useMemo, useState } from 'react'
```

Comment:
```suggestion
import * as React from 'react'
```

What do you think about this? Actually, this is the only right way to import React as ESM module. Moreover it is much easier to read code when default react hooks are started with `React.` 

https://github.com/facebook/react/pull/18102
https://twitter.com/sebmarkbage/status/1231673639080038400

Reply:
> Actually, this is the only right way to import React as ESM module

Is this actually relevant to us? Why would it matter?

> much easier to read code when default react hooks are started with `React.`

I disagree with this and see no significant difference other than styling. I will not be changing it for your (CT) codebase, but for `design-system` I will continue importing React as above.

Reply:
I did post 2 links. You are doing actually an invalid thing, you are trying to import UMD using esm default export. Once react will finally have esm export this will not be possible. So **it's not the styling**.

Reply:
I know very little about the crazy mess that is different types of JS modules, but the most recent React team interaction on modules/importing was the following: https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports

In particular, I will point out this:

> Change all default React imports (i.e. import React from "react") to destructured named imports (ex. import { useState } from "react") which is the preferred style going into the future. This codemod will not affect the existing namespace imports (i.e. import * as React from "react") which is also a valid style. The default imports will keep working in React 17, but in the longer term we encourage moving away from them.

As a question of my own, don't `import *` imports hurt tree shaking?

Reply:
> As a question of my own, don't import * imports hurt tree shaking?

No, just because for now it's still an all-piece of react in one object that just exports different parts. But yeah export default is the biggest mess.